backend/src/main/java/fk/prof/backend/util/ProtoUtil.java

```diff
@@ -27,4 +23,10 @@ public static Buffer buildBufferFromProto(AbstractMessage message) throws IOExce
     byteBuf.writerIndex(serializedSize);
     return Buffer.buffer(byteBuf);
   }
+
+  //Proto parser operates directly on underlying byte array, avoids byte copy
+  public static <T extends AbstractMessage> T buildProtoFromBuffer(Parser<T> parser, Buffer buffer)
+      throws InvalidProtocolBufferException {
+    return parser.parseFrom(CodedInputStream.newInstance(buffer.getByteBuf().nioBuffer()));
```

Comment:
The returned proto object can be used further down the line, across threads, and its longevity is not known to this method. Will discuss offline to better understand the comment